tests/features/camel-as2/src/main/java/org/apache/karaf/camel/test/CamelAs2RouteSupplier.java

```diff
@@ -0,0 +1,130 @@
+            + "IMD+F++:::DIFFERENT WIDGET'\n"
+            + "QTY+47:20:EA'\n"
+            + "ALI+JP'\n"
+            + "MOA+203:410'\n"
+            + "PRI+INV:20.5'\n"
+            + "UNS+S'\n"
+            + "MOA+39:2137.58'\n"
+            + "ALC+C+ABG'\n"
+            + "MOA+8:525'\n"
+            + "UNT+23+00000000000117'\n"
+            + "UNZ+1+00000000000778'\n";
+    private static final String EDI_MESSAGE_CONTENT_TRANSFER_ENCODING = "7bit";
+
+    @Override
+    public void configure(CamelContext context) {
+        final int port = Integer.parseInt(System.getProperty("as2.port"));
+        final AS2Configuration configuration = new AS2Configuration();
+        configuration.setTargetHostname("localhost");
+        configuration.setTargetPortNumber(port);
+        configuration.setServerFqdn("example.com");
+        configuration.setServerPortNumber(port);
+
+        final AS2Component component = new AS2Component(context);
+        component.setConfiguration(configuration);
+        context.addComponent("as2", component);
+    }
+
+
+    @Override
+    protected Function<RouteBuilder, RouteDefinition> consumerRoute() {
+        return builder ->
+                builder.from("as2://server/listen?requestUriPattern=/")
+                        .log("received message ${body}")
+                        .setBody(constant("OK"))
+                        ;
+    }
+
+    @Override
+    protected void configureProducer(RouteBuilder builder, RouteDefinition producerRoute) {
+        producerRoute.setHeader("CamelAS2.requestUri",constant(REQUEST_URI))
+                .setHeader("CamelAS2.subject",constant(SUBJECT))
+                .setHeader("CamelAS2.from", constant(FROM))
+                .setHeader("CamelAS2.as2From", constant(AS2_NAME))
+                .setHeader("CamelAS2.as2To", constant(AS2_NAME))
+                .setHeader("CamelAS2.as2MessageStructure", constant(AS2MessageStructure.PLAIN))
+                .setHeader("CamelAS2.ediMessageContentType",
+                        constant(ContentType.create(AS2MediaType.APPLICATION_EDIFACT, StandardCharsets.US_ASCII.name())))
+                .setHeader("CamelAS2.ediMessageTransferEncoding", constant(EDI_MESSAGE_CONTENT_TRANSFER_ENCODING))
+                .setHeader("CamelAS2.dispositionNotificationTo", constant(FROM))
+                .setHeader("CamelAS2.attachedFileName", constant(""))
+                .setBody(constant(EDI_MESSAGE))
+                .to("as2:client/send?inBody=ediMessage")
+                .log("message sent successfully");
+    }
+
+    @Override
+    public void createRoutes(RouteBuilder builder) {
+        if (consumerEnabled()) {
+            configureConsumer(consumerRoute().apply(builder));
+        }
+        if (producerEnabled()) {
+            configureProducer(
+                    builder, builder.fromF("direct:%s", getTestComponentName()).routeId("producer-%s".formatted(getTestComponentName()))
+            );
+        }
```

Comment:
Could you please clarify why it is needed?

Reply:
when the producer starts before the consumer, it fails with a connection exception
` because of Failed to resolve endpoint: as2://client/send?inBody=ediMessage due to: Client HTTP connection failed `
I tried the 'lazyStartProducer' option but no effect. That's why I inverted the logic in the overriden method. I don't think it's the general rule though, so I left it as a override. I can add a comment

Reply:
if so, yes please add a comment and you can also remove the if blocks as they are useless in this case since it always be true